packages/fetchai/connections/soef/connection.py

```diff
@@ -165,6 +165,13 @@ def register_service(self, service_description: Description) -> None:
                 )
             )
 
+    def disconnect(self):
```

Comment:
Only the service referenced in the data model should be disconnected.

Reply:
So the functionality you would like it to have is that the skill calls the disconnect part ? Otherwise how I would know the service name if we don't store it ? How do you get reference of the data_model inside the connection?

Reply:
We do store the service name, that's the point. Have a look at the remaining implementation.